analyses/sex-prediction-from-RNASeq/04-present_results.Rmd

```diff
@@ -0,0 +1,249 @@
+---
+title: "Sex Prediction from RNASeq Results"
+output: html_document
+author: Bill Amadio
+date: 2019
+params:
+  results_dir: "results"
+  model_dir: "models"
+  cm_set: NULL
+  seed: NULL
+  target_columns: NULL
+---
+
+```{r echo=FALSE, results='hide'}
+
+library(dplyr, warn.conflicts=FALSE, quietly = TRUE)
+library(readr)
+library(ggplot2)
+library(graphics)
+library(rmarkdown)
+library(knitr)
+
+```
+
+
+```{r echo=FALSE, results='hide'}
+#in this chunk, we read the columns required for the predictive accuracy v. # training transcripts and
+## non-zero features v. # training transcripts plote
+
+# we will generate a separate data frame for each value in the targetColumns argument
+target_column_list <- strsplit(params$target_columns, ",")
+df_list <- list()
+
+# Create specified output directory if it does not yet exist
+plot_directory <- file.path(params$results_dir, "plots")
+
+if (!dir.exists(plot_directory)) {
+  dir.create(plot_directory, recursive = TRUE)
+}
+
+#loop through the values in the targetColumns argument
+for (t in target_column_list[[1]]) {
+
+# here, we collect the predictive accuracy values from each confusion matrix file in the results directory  
+  results_directory <- file.path(params$results_dir, t)
+  files_to_process <- list.files(path = results_directory, pattern = "_confusion_matrix.RDS")
+  first_pos <- sapply(files_to_process, function(x) regexpr(params$seed, x))
+  last_pos <- sapply(files_to_process, function(x) regexpr("confusion_matrix", x))
+  transcript_tail_percents <- substr(files_to_process, 1, last_pos-2)
```

Comment:
I will look at how to use tidyverse and stringr here.

Reply:
Everything seems to be _correct_. I am running the whole pipeline now so I can take another look locally. I want to get this merged so that the module has a README in the master branch and then any improvements of this nature can go into a subsequent pull request. In general, we aim to keep pull requests focused on a single file or area per [the contributing guidelines](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/CONTRIBUTING.md#size-and-composition-of-pull-requests) which helps ensure a thorough review can be completed in a timely manner. So I think this is a case where we want to get everything you've done in and then we can refine things as needed.